### DISCUSSION TITLE
dumi2兼容react-router-dom5

### DISCUSSION THREAD

**Discussion by Grace-JingXiao** [umijs/dumi#1834] [Q&A]
dumi2能兼容react-router-dom5么, 怎么配置，旧项目是react16+react-router v5,

**Comment by PeachScript** [comment]
dumi 的自身的 router 是 v6，没有降级方案；对于在 dumi 里使用 react-router 有两点建议：
1. 组件库应该是和路由解耦的，router 由应用决定，所以理论上使用 dumi 开发的组件库不需要关心 router
2. 如果要以 router 作为示例来展示（比如 antd 的 Breadcrumb 组件），那么可以自己在 demo 里用 createRoot 的形式重新创建一个 React 应用，该应用和 dumi 是两回事不受 dumi 的 router 版本影响

**Reply by Grace-JingXiao** [reply]
我们的场景不纯粹是组件，有些是跟业务相关的，是一些场景相关的封装，所以内部一些封装会依赖router。如何在demo里注入router5的依赖，最好是对全局的demo注入router5，有相关的插件和api么
